web3js-ext/test/role_based_key_feepayer.spec.ts

```diff
@@ -83,36 +83,73 @@ describe("Role-based Key Tests", function () {
         }
     });
 
-    // Test Case 3: Fee Delegated transaction signed by RoleFeePayer key (should succeed)
     it("3. Fee Delegated transaction signed by RoleTransaction and RoleFeePayer keys", async function () {
-        // Feedback 4. Create a new user account (different user than FeePayer)
         const userAccount = generateTemporaryAccount();
-
+    
         const feeDelegatedTx = {
             type: TxType.FeeDelegatedValueTransfer,
             from: userAccount.address,
             to: generateTemporaryAccount().address,
             value: toPeb("0.01"),
-            gasLimit: "100000", 
-            gasPrice: "25000000000", 
+            gasLimit: "100000",
+            gasPrice: "25000000000",
             nonce: "0x0",
-            chainId: "0x1001" 
+            chainId: "0x1001"
         };
-
-        // Feedback 4. User signs transaction first
+    
+        // 1) A User signs a transaction
         const signedTxByUser = await userAccount.signTransaction(feeDelegatedTx);
         assert.isNotNull(signedTxByUser.rawTransaction, "Transaction should be signed by User");
-
-        // Feedback 4. FeePayer adds an additional signature
+    
+        console.log("Signed Transaction by User (rawTransaction):", signedTxByUser.rawTransaction);
+    
         try {
-            const signedTxByFeePayer = await roleFeePayerAccount.signTransaction({
+            // 2) FeePayer signs a transaction
+            const feePayerSignInput = {
+                type: feeDelegatedTx.type,
+                from: feeDelegatedTx.from,
+                to: feeDelegatedTx.to,
+                value: feeDelegatedTx.value,
+                gasPrice: feeDelegatedTx.gasPrice,
+                gasLimit: feeDelegatedTx.gasLimit,
+                nonce: feeDelegatedTx.nonce,
+                chainId: feeDelegatedTx.chainId,
+                feePayer: roleFeePayerAccount.address,
                 senderRawTransaction: signedTxByUser.rawTransaction,
-                gasPrice: "25000000000",
-                gasLimit: "100000" 
-            });
+            };
+            console.log("FeePayer SignTransaction Input:", feePayerSignInput);
+    
+            const signedTxByFeePayer = await roleFeePayerAccount.signTransaction(feePayerSignInput);
             assert.isNotNull(signedTxByFeePayer.rawTransaction, "FeePayer should sign the transaction");
+    
+            console.log("Signed Transaction by FeePayer (rawTransaction):", signedTxByFeePayer.rawTransaction);
+    
+            // 3) RLP format check
+            if (!signedTxByFeePayer.rawTransaction.startsWith("0x")) {
+                console.error("Invalid rawTransaction format");
+            } else {
+                // 4) RLP Decoding
+                const { KlaytnTxFactory } = require("@kaiachain/web3js-ext");
+                const decoded = KlaytnTxFactory.fromRLP(signedTxByFeePayer.rawTransaction);
+                console.log("Decoded Transaction:", decoded);
```

Comment:
@kjeom 
**[What I Tried]**
1. Decoding (RLP) to Verify FeePayer Address
- I attempted to check whether the actual FeePayer address matches my intended address by using KlaytnTxFactory.fromRLP(...) after signing the transaction twice (User first, then roleFeePayerAccount).
- Once both signatures were complete, I decoded the final raw transaction, expecting to see the FeePayer field populated with my intended FeePayer address.
2. Observation
- However, after decoding, I found that the feePayer field was empty, suggesting that the FeePayer signature wasn’t reflected in the final transaction.
![image](https://github.com/user-attachments/assets/bfbfb308-fc9d-4431-8b23-1260793069bd)

**[Question]**
“If the feePayer field is empty in the decoded transaction, does that imply our two-step signing process is incorrect, or could there be a bug in how web3js-ext handles RoleBased FeePayer signatures?”

More specifically:
- I called signTransaction twice (User → FeePayer), providing all necessary fields (type, from, to, value, gasLimit, gasPrice, nonce, chainId, feePayer, and senderRawTransaction).
- Despite both signings appearing to succeed, the decoded result shows no FeePayer data.

**[Additional Notes]**
- This issue arises when using RoleBased accounts for FeeDelegated transactions. Even though the second signing step looks successful (i.e., I get a new raw transaction), the final RLP does not contain the feePayer field.
- I'm curious if this is related to a known RoleBased + FeeDelegated limitation or bug in web3js-ext...?